neo/Ledger/BlockState.cs

```diff
@@ -6,44 +6,38 @@ namespace Neo.Ledger
 {
     public class BlockState : StateBase, ICloneable<BlockState>
     {
```

Comment:
`BlockState` will now just have the TrimmedBlock.
Can we rename or simplify?

Reply:
How to simplify?

Reply:
Not sure, couldn't it be merged inside TrimmedBlock itself? It would be a little bit strange to inherited `StateBase`, right?

But is this `StateBase` really useful? It just have `StateVersion=0`.
I thought if all this could be abolished.

Reply:
@shargon @igormcoelho, what do you think?

Reply:
I think this is feasible, but we should make this change in a separate PR.